wip-Gorka-Irazoqui-ARS-merkelization.md

```diff
@@ -0,0 +1,173 @@
+
+Howver, for the smart contract to determine the validity of the block header it first needs to validate that the signature was signed by a legitimate ARS member. At the moment the smart contract platform does not posses access to such information, as it only resides in the Witnet chain.
+
+In other words, the smart contract storing the block headers requires the ability to validate Proof of Memberships. The bridge nodes in this case are in charge of generating the proofs each ARS member they want to add a vote for. Nevertheless, such a scheme requires some kind of information about the ARS to be stored together with the block header.
+
+This proposal describes how to achieve Proof of Memberships by merkelizing the ARS and appending the root to the block header. With such a scheme, demonstrating that a particular identity belongs to the ARS is as easy as providing the merkle path leading to the merkle root. In the following sections we refer to block headers as superblocks, which are simply groups of block headers that are join together in one single superblock block header.
+
+## Specification
+### Forming a ARS with public keys in G2
+The current implementation of BLS signatures in the node assumes that the verification to be done in Ethereum is *e(S, G<sub>2</sub>) = e(H(M), P)*, where:
+
+- *S* is an individual/aggregated signature in G1
+- *G<sub>2</sub>* is the generator in the field G2
+- *H(M)* is the output of a hash to point function that transforms a message *M* and outputs a point in G1.
+- *P* is an individual/aggregated public key in G2.
+
+Therefore, the public keys to be aggregated are those in G2, and that is exactly the reason why the public keys in G2 need to be committed alongside eligibility proofs. With those public keys in G2, the ARS is composed of the public keys of ARS members in G2.
+
+Each of the public keys is composed of 65 bytes in compressed form, and 128 bytes in uncompressed form. Now the question that follows is, should we use the compressed or the uncompressed form to from the merkle tree? To answer the question, we should take a look at how we can calculate the uncompressed from the compressed form. In the case of G2, the uncompressed form calculation involves complex operations in G2. Note that these public keys will need to be provided to the smart contract verifying the ARS membership. Thus, for the moment we decided that such computations are too complex to perform in the Ethereum EVM.
```

Comment:
```suggestion
Each of the public keys is composed of 65 bytes in compressed form, and 128 bytes in uncompressed form. Now the question that follows is, should be used the compressed or the uncompressed form in the merkle tree? It will depend  on how the calculation of the uncompressed and the compressed form in done. In the case of G2, the uncompressed form calculation involves complex operations in G2. Note that these public keys will need to be provided to the smart contract verifying the ARS membership. Thus, for the moment it has been decided that such computations are too complex to perform in the Ethereum EVM.
```